zookeeper/core/component_test.py

```diff
@@ -646,6 +647,64 @@ class Parent:
     assert instance.child.a == 5
 
 
+def test_component_configure_override():
+    @component
+    class A:
+        def __configure__(self, conf, **kwargs):
+            pass
+
+    # It doesn't call `configure`, so an error should be thrown.
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "`A` remains unconfigured after calling __configure__! Make sure to call"
+            " `configure(self, conf, **kwargs)` at the end of this function."
+        ),
+    ):
+
+        instance = A()
+        instance.__configure__({})
+
+    # This should pass
+    @component
+    class B:
+        def __configure__(self, *args, **kwargs):
+            configure(self, *args, **kwargs)
+
+    instance = B()
+
+    # This should not, since it doesn't take any additional args
+    with pytest.raises(
+        TypeError,
+        match=re.escape(
+            "The `__configure__` method of a @component class must match the arguments "
+            "of `configure()`, but `C.__configure__` accepts arguments ('self',). "
+            "Valid arguments: (self, conf, name, interactive, *args, **kwargs)"
+        ),
+    ):
+
+        @component
+        class C:
+            def __configure__(self):
+                pass
+
+        instance = C()
```

Comment:
I realise now that this test will fail if we go with my suggestion regarding error messages. However, I still think it's a good idea to simply the error message handling, and therefore remove this test-case, since it's very complicated to cover all cases (checking that the function signatures of the two methods are exactly compatible) and there's a danger of false-positive error messages. Plus, if the signature of `__configure__` is wrong it'll throw during configure, which happens very early on.

Reply:
Yes fair point. All should be fixed in 268a117